kbase-extension/static/kbase/js/widgets/narrative_core/upload/importSetup.js

```diff
@@ -94,15 +164,26 @@ define(['bluebird', 'base/js/namespace', 'narrativeConfig', 'util/kbaseApiUtil',
                     };
                 }
                 bulkFiles[importType].files.push(file.name);
-            } else if (importType === 'csv/tsv/spreadsheet/whatever') {
+            } else if (importType === 'import_specification') {
                 xsvFiles.push(file.name);
             } else {
                 singleFiles.push(file);
             }
         });
         return getSpreadsheetFileInfo(xsvFiles)
-            .then(() => {
-                // TODO: merge xsvFileInfo with bulkFiles, handle errors, etc.
+            .then((result) => {
+                if (result.types) {
+                    Object.keys(result.types).forEach((dataType) => {
+                        if (!(dataType in bulkFiles)) {
+                            bulkFiles[dataType] = {
+                                appId: uploaders.app_info[dataType].app_id,
+                                files: [],
+                                outputSuffix: uploaders.app_info[dataType].app_output_suffix,
+                            };
+                        }
+                        bulkFiles[dataType].appParameters = result.types[dataType];
```

Comment:
`appParameters` is the set of data in the xSV file. This just gets added to the set of data sent to the bulkImportCell on startup.